### PR TITLE
Add some exercises to write tests

### PR DESCRIPTION
**Pull request by 40thieves**
Adds some exercises to get trainees to write tests (instead of just using the output of tests).

I've split into a simpler example with a bit more scaffolding, and a more advanced example with a bit more complexity.

I'll leave this as a draft PR for now, as I think we may need to add supporting content to the syllabus around how to write tests. @CodeYourFuture/global-syllabus is due to discuss TDD soon, so this is more of a proposal for discussion.

### CODE REVIEW

**Comment by 40thieves** [comment]
Discussed in global syllabus, we think it's worth going ahead with this --- people can always fix it if they don't like it :)

**Comment by 40thieves** [comment]
Assigned some of the folks who were at the TDD call the other week, but I'm afraid I don't know Tom's handle --- so feel free to point him this way if you know 🙂

File: 2-mandatory/5-writing-tests.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+});
+
+/*
+  The rest of the tests have comments describing what to test and you need to
+  write a matching test
+*/
+
+/*
+  Write a test that checks a score of 71 is grade B
+*/
```
Comment (textbook) [inline 2-mandatory/5-writing-tests.js]: I wonder if the first of these could be provided with slightly more structure to ease into writing everything themselves:
```suggestion
test.skip("a score of 71 is grade B", () => {
  /* Write the test body */
});
```

**Comment by SallyMcGrath** [comment]
> 
Tom is @itsravenous 👋 🦖

**Review by Dedekind561 — APPROVED** [review]
You can make a ticket to come back and review this piece of work - but let's get merged for now

**Comment by SallyMcGrath** [comment]
@40thieves let's merge this now?

File: 2-mandatory/5-writing-tests.js (hunk 1 above)
Comment (alismith-dev) [reply 2-mandatory/5-writing-tests.js]: Thanks. I've accepted this, plus added instructions about removing the `.skip`.

**Comment by 40thieves** [comment]
> a ticket to come back and review this piece of work - but let's get merged for now

Makes sense, thanks.

**Comment by 40thieves** [comment]
@textbook @itsravenous if you'd like to review/update these tests, could you open a ticket to describe them in detail? Thanks 🙂

**Comment by sztupy** [comment]
We just had a chat about this, and we believe exercise 6 could have gotten into the `extra` folder. We already face some issues with trainees having too much mandatory coursework

**Pull request merged**